clients/web/src/views/widgets/SearchFieldWidget.js

```diff
@@ -195,7 +197,10 @@ var SearchFieldWidget = View.extend({
             data: {
                 q: q,
                 mode: this.currentMode,
-                types: JSON.stringify(this.types)
+                types: JSON.stringify(_.intersection(
+                    this.types,
+                    SearchFieldWidget.getModeTypes(this.currentMode))
+                )
```

Comment:
Performing an intersection here breaks search results when `SearchFieldWidget` is initialized with a custom type, because `types` will be empty.